lib/krane/deprecated_deploy_task.rb

```diff
@@ -125,7 +125,7 @@ def server_version
     # @param protected_namespaces [Array<String>] Array of protected Kubernetes namespaces (defaults
     #   to Krane::DeployTask::PROTECTED_NAMESPACES)
     # @param render_erb [Boolean] Enable ERB rendering
-    def initialize(namespace:, context:, current_sha:, logger: nil, kubectl_instance: nil, bindings: {},
+    def initialize(namespace:, context:, current_sha: nil, logger: nil, kubectl_instance: nil, bindings: {},
       max_watch_seconds: nil, selector: nil, template_paths: [], template_dir: nil, protected_namespaces: nil,
       render_erb: true, allow_globals: false)
       template_dir = File.expand_path(template_dir) if template_dir
```

Comment:
At some point we call `TemplateSets::from_dirs_and_files` on the input, what do we do if there are `.erb` files included in the set? We can either ignore (I think that's wrong) or raise (fast, before waiting for a downstream error). Unfortunately, the library code still supports `current_sha` and `bindings`, so it _is_ still valid to pass in ERB... do we need to scan at the CLI command level for appropriate templates, at least in the short term?

Now that `deploy` doesn't render, there's no longer a 1-1 correspondence between it and the `render` task as far as `TemplateSets` goes.

Reply:
So this isn't a new failure case, you can set `render_erb: false` and pass in erb templates. The don't get rendered and will fail likely fail on apply unless somehow your erb is valid yaml&k8s. 

I agree this is something we should fail fast on, the right path looks like passing `render_erb` to the TemplateSet#validtate  and raising there. 🤞 that doesn't cause test failures.

Reply:
Error messaging is coming in https://github.com/Shopify/kubernetes-deploy/pull/635/files